tests/integration/_common.py

```diff
@@ -206,23 +212,72 @@ def __enter__(self):
         # Figure out where the tox-initiated venv is. In environments where a
         # venv is activated as part of bashrc, $VIRTUAL_ENV won't be what we
         # expect inside of tmux, so we can't depend on it. We *must* determine
-        # it before we enter tmux.
+        # it before we enter tmux. Do this before we switch to bash
         venv_path = os.environ.get("VIRTUAL_ENV")
         if venv_path is None or ".tox" not in venv_path:
             raise AssertionError(
                 "VIRTUAL_ENV environment variable was not set but tox should have set it."
             )
         venv = os.path.join(shlex.quote(venv_path), "bin", "activate")
+        # get the USER before we start a clean shell
+        user = os.environ.get("USER")
+        home = os.environ.get("HOME")
+
+        # get a clean shell and predicatble prompt
+        self._cli_prompt = self._get_cli_prompt()
+
+        # set envars for this session
+        tmux_common = [f". {venv}"]
+        tmux_common.append(f"export HOME={home}")
+        tmux_common.append(f"export USER={user}")
+        tmux_common.append(f"export ANSIBLE_NAVIGATOR_CONFIG={self._config_path}")
+        tmux_common.append("export ANSIBLE_NAVIGATOR_LOG_LEVEL=debug")
+
+        self._test_log_dir = None
+        if user == "zuul":
+            self._test_log_dir = os.path.join(
+                "/home/zuul", "zuul-output", "logs", "anible-navigator", self._test_path
+            )
+        else:
+            self._test_log_dir = os.path.join("./", ".test_logs", self._test_path)
 
-        # send the config envvar + other set up commands
-        venv = f"source {venv}"
-        navigator_config = f"export ANSIBLE_NAVIGATOR_CONFIG={self._config_path}"
-        set_up_commands = [venv, navigator_config] + self._setup_commands
+        os.makedirs(self._test_log_dir, exist_ok=True)
+
+        log_file = os.path.join(self._test_log_dir, "ansible-navigator.log")
+        tmux_common.append(f"export ANSIBLE_NAVIGATOR_LOG_FILE={log_file}")
+        playbook_artifact = os.path.join(self._test_log_dir, "playbook-artifact.log")
+        tmux_common.append(
+            f"export ANSIBLE_NAVIGATOR_PLAYBOOK_ARTIFACT_SAVE_AS={playbook_artifact}"
+        )
+        collection_doc_cache = os.path.join(self._test_log_dir, "collection_doc_cache.db")
+        tmux_common.append(
+            f"export ANSIBLE_NAVIGATOR_COLLECTION_DOC_CACHE_PATH={collection_doc_cache}"
+        )
+
+        set_up_commands = tmux_common + self._setup_commands
         set_up_command = " && ".join(set_up_commands)
+
+        # send setup, wait for the prompt in last line
```

Comment:
Can we return early and fail out somehow if the setup commands fail for any reason? They're `&&`'d so it'll give us back a useful return code, but that's not really useful since we're just screen-scraping. If they fail, we'll still sit around waiting. :thinking:

Reply:
added a timer for the prompt after, and we'll log the setup screen